src/physics/utils/phys_comp.F90

```diff
@@ -237,32 +235,35 @@ subroutine phys_run2()
 
    end subroutine phys_run2
 
-   subroutine phys_timestep_final()
+   subroutine phys_timestep_final(do_ncdata_check)
       use time_manager,   only: get_nstep
       use cam_abortutils, only: endrun
       use cam_initfiles,  only: unset_path_str
       use cam_ccpp_cap,   only: cam_ccpp_physics_timestep_final
       use physics_inputs, only: physics_check_data
 
+      ! Subroutine inputs
+      logical, intent(in) :: do_ncdata_check
+
       ! Local variables
-      integer                              :: data_frame
+      integer             :: data_frame
 
       ! Finalize the time step
       call cam_ccpp_physics_timestep_final(phys_suite_name)
       if (errcode /= 0) then
          call endrun('cam_ccpp_physics_timestep_final: '//trim(errmsg))
       end if
 
-      ! data_frame is the next input frame for physics input fields
-      ! Frame 1 is skipped for snapshot files
-      !!XXgoldyXX: This section needs to have better logic once we know if
-      !!           this is a physics test bench run.
-      data_frame = get_nstep() + 2
+      ! data_frame is the next input frame for
+      ! physics snapshot validation fields
+      data_frame = get_nstep()
 
       ! Determine if physics_check should be run:
       if (trim(ncdata_check) /= trim(unset_path_str)) then
-         call physics_check_data(ncdata_check, suite_names, data_frame,       &
-            min_difference, min_relative_value)
+         if (do_ncdata_check) then
```

Comment:
do we still want to include the check if ncdata_check is populated in order to proceed to the check? I see that there's logic in physics_check data to issue a warning ('WARNING: Namelist variable ncdata_check is UNSET. Model will run, but physics check data will not be printed') if it's not, but I can also imagine that warning would be confusing if you had no intention of running the check.

Reply:
That's a good question!  I guess my opinion would be to just skip the call to `physics_check_data` entirely if we can avoid it, so I would prefer to leave this `ncdata_check` if-statement here as-is.

That being said, I am also not sure if the extra internal logic inside of `physics_check_data` you mentioned is really hurting anything either (beyond maybe a few wasted clock cycles).  So that combined with the fact that I am not totally sure what the long-term plan for the `physics_check_data` routine will be (e.g. will it eventually be called elsewhere in SIMA?  Removed entirely?) makes me lean towards just leaving everything as-is on that front for now as well.  Of course if you or @cacraigucar feel strongly one way or the other please let me know.  Thanks!

Reply:
I like avoiding writing out the WARNING message for runs where we are not performing checks (normal CAM-SIMA runs in the future?)  I figure when we get to that stage, we can make sure that everything is running "best" for CAM-SIMA and make any required tweeks at that time.  So I don't have a strong feeling one way or the other.

Reply:
Well, turns out I can't read. The `if (trim(ncdata_check) /= trim(unset_path_str)) then` is still there and I thought it was gone. So.. agreed and nevermind!